In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np

from math import nan

pd.set_option('display.max_columns', None)

## Combining data from the different sources in one dataframe

In [ ]:
paths = ['../data/processed/1082ff_monatsbericht_suesswaren_schoko_kaffee_processed.csv', 
         '../data/processed/1082ff_erzeugerpreise_suesswaren_schoko_kaffee_processed.csv', 
         '../data/processed/1082_1083_einfuhrpreise_suesswarenundkaffee_processed.csv', 
         '../data/processed/01183ff_vpi_suesswaren_schoko_kaffee_processed.csv', 
         '../data/processed/A1014_17_18_monprod_suesswarenequi_kaffee_processed.csv', 
         '../data/processed/globalcacaoprice_fred_processed.csv', 
         '../data/processed/globalcoffeerobustaprice_fred_processed.csv']
# Note that '../data/processed/1082_monatsbericht_suesswaren_timeseries_processed.csv'
# contains a subset of the information of 
# '../data/processed/1082ff_monatsbericht_suesswaren_schoko_kaffee_processed.csv'

In [ ]:
# Read all the dataframes from the given paths
dfs = []
for path in paths:
    df_help = pd.read_csv(path, index_col = 'date')
    dfs.append(df_help)
# Concatenate the dataframes
dfs = pd.concat(dfs, axis=1)
print(dfs.shape)

../data/processed/1082ff_monatsbericht_suesswaren_schoko_kaffee_processed.csv
../data/processed/1082ff_erzeugerpreise_suesswaren_schoko_kaffee_processed.csv
../data/processed/1082_1083_einfuhrpreise_suesswarenundkaffee_processed.csv
../data/processed/01183ff_vpi_suesswaren_schoko_kaffee_processed.csv
../data/processed/A1014_17_18_monprod_suesswarenequi_kaffee_processed.csv
../data/processed/globalcacaoprice_fred_processed.csv
../data/processed/globalcoffeerobustaprice_fred_processed.csv


In [4]:
dfs = pd.concat(dfs, axis=1)
print(dfs.shape)
        

(84, 167)


In [5]:
dfs

Betriebe_WZ08-10 Betriebe_ep_WZ08-10  Beschäftigte_WZ08-10  \
date                                                                     
2019-01-31            2394.0                   e              449679.0   
2019-02-28            2424.0                   e              451567.0   
2019-03-31            2438.0                   e              452452.0   
2019-04-30            2434.0                   e              452305.0   
2019-05-31            2426.0                   e              451864.0   
...                      ...                 ...                   ...   
2025-08-31            2403.0                   p              514261.0   
2025-09-30            2401.0                   p              510487.0   
2025-10-31            2397.0                   p              510214.0   
2025-11-30            2395.0                   p              514850.0   
2025-12-31               NaN                 NaN                   NaN   

           Beschäftigte_ep_WZ08-10  Geleistete Arbeitsstunden_WZ08-10  \
date                                                                    
2019-01-31                       e                         60212000.0   
2019-02-28                       e                         56795000.0   
2019-03-31                       e                         58455000.0   
2019-04-30                       e                         57711000.0   
2019-05-31                       e                         59098000.0   
...                            ...                                ...   
2025-08-31                       p                         62492000.0   
2025-09-30                       p                         64618000.0   
2025-10-31                       p                         65436000.0   
2025-11-30                       p                         64454000.0   
2025-12-31                     NaN                                NaN   

           Geleistete Arbeitsstunden_ep_WZ08-10  \
date                                              
2019-01-31                                    e   
2019-02-28                                    e   
2019-03-31                                    e   
2019-04-30                                    e   
2019-05-31                                    e   
...                                         ...   
2025-08-31                                    p   
2025-09-30                                    p   
2025-10-31                                    p   
2025-11-30                                    p   
2025-12-31                                  NaN   

            Bruttolohn- und -gehaltssumme_WZ08-10  \
date                                                
2019-01-31                           1.214597e+09   
2019-02-28                           1.174794e+09   
2019-03-31                           1.242076e+09   
2019-04-30                           1.253613e+09   
2019-05-31                           1.286052e+09   
...                                           ...   
2025-08-31                           1.698105e+09   
2025-09-30                           1.697682e+09   
2025-10-31                           1.758717e+09   
2025-11-30                           2.234452e+09   
2025-12-31                                    NaN   

           Bruttolohn- und -gehaltssumme_ep_WZ08-10  Umsatz_WZ08-10  \
date                                                                  
2019-01-31                                        e    1.168068e+10   
2019-02-28                                        e    1.106253e+10   
2019-03-31                                        e    1.191681e+10   
2019-04-30                                        e    1.234662e+10   
2019-05-31                                        e    1.208184e+10   
...                                             ...             ...   
2025-08-31                                        p    1.542404e+10   
2025-09-30                                        p    1.664659e+10   
2025-10-31                         

In [6]:
# transforming dates from first day of month to last day of month

# importing date transformation functions
import calendar
from datetime import datetime

# base for for loop:
date_list = list(dfs.index)
print(date_list[0])
date_list = [datetime.strptime(date, '%Y-%m-%d') for date in date_list]
input_dt = date_list[0]
print("The original date is:", input_dt.date())

# monthrange() to get the date range
res = calendar.monthrange(input_dt.year, input_dt.month)
day = res[1]
print(f"Last date of month is: {input_dt.year}-{input_dt.month}-{day}")


2019-01-31
The original date is: 2019-01-31
Last date of month is: 2019-1-31


In [7]:
# transforming all dates in the index to the last day of the month
new_index_list = []
for dates in date_list:
    res = calendar.monthrange(dates.year, dates.month)
    day = res[1]
    new_date = f"{dates.year}-{dates.month}-{day}"
    new_index_list.append(new_date)
new_index_list = [pd.to_datetime(date).strftime('%Y-%m-%d') for date in new_index_list]


# applying new index to dfs:
dfs.index = new_index_list
#print(dfs.index)

In [9]:
path = '../data/processed/shareprice_monthly.csv'
df_help = pd.read_csv(path, index_col = 'date', skipfooter=1)

dfs = pd.concat([dfs, df_help], axis=1)
dfs.index.name = 'date'
dfs

C:\Users\judit\AppData\Local\Temp\ipykernel_21896\1903041209.py:2: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support skipfooter; you can avoid this warning by specifying engine='python'.
  df_help = pd.read_csv(path, index_col = 'date', skipfooter=1)


Betriebe_WZ08-10 Betriebe_ep_WZ08-10  Beschäftigte_WZ08-10  \
date                                                                     
2019-01-31            2394.0                   e              449679.0   
2019-02-28            2424.0                   e              451567.0   
2019-03-31            2438.0                   e              452452.0   
2019-04-30            2434.0                   e              452305.0   
2019-05-31            2426.0                   e              451864.0   
...                      ...                 ...                   ...   
2025-08-31            2403.0                   p              514261.0   
2025-09-30            2401.0                   p              510487.0   
2025-10-31            2397.0                   p              510214.0   
2025-11-30            2395.0                   p              514850.0   
2025-12-31               NaN                 NaN                   NaN   

           Beschäftigte_ep_WZ08-10  Geleistete Arbeitsstunden_WZ08-10  \
date                                                                    
2019-01-31                       e                         60212000.0   
2019-02-28                       e                         56795000.0   
2019-03-31                       e                         58455000.0   
2019-04-30                       e                         57711000.0   
2019-05-31                       e                         59098000.0   
...                            ...                                ...   
2025-08-31                       p                         62492000.0   
2025-09-30                       p                         64618000.0   
2025-10-31                       p                         65436000.0   
2025-11-30                       p                         64454000.0   
2025-12-31                     NaN                                NaN   

           Geleistete Arbeitsstunden_ep_WZ08-10  \
date                                              
2019-01-31                                    e   
2019-02-28                                    e   
2019-03-31                                    e   
2019-04-30                                    e   
2019-05-31                                    e   
...                                         ...   
2025-08-31                                    p   
2025-09-30                                    p   
2025-10-31                                    p   
2025-11-30                                    p   
2025-12-31                                  NaN   

            Bruttolohn- und -gehaltssumme_WZ08-10  \
date                                                
2019-01-31                           1.214597e+09   
2019-02-28                           1.174794e+09   
2019-03-31                           1.242076e+09   
2019-04-30                           1.253613e+09   
2019-05-31                           1.286052e+09   
...                                           ...   
2025-08-31                           1.698105e+09   
2025-09-30                           1.697682e+09   
2025-10-31                           1.758717e+09   
2025-11-30                           2.234452e+09   
2025-12-31                                    NaN   

           Bruttolohn- und -gehaltssumme_ep_WZ08-10  Umsatz_WZ08-10  \
date                                                                  
2019-01-31                                        e    1.168068e+10   
2019-02-28                                        e    1.106253e+10   
2019-03-31                                        e    1.191681e+10   
2019-04-30                                        e    1.234662e+10   
2019-05-31                                        e    1.208184e+10   
...                                             ...             ...   
2025-08-31                                        p    1.542404e+10   
2025-09-30                                        p    1.664659e+10   
2025-10-31                         

In [19]:
columns_without_space = []
for col in dfs.columns.values:
    columns_without_space.append(col.replace("  ", ""))
dfs.columns = columns_without_space

In [21]:
# saving combined data to csv
dfs.to_csv('../data/processed/combined_data.csv')
pd.DataFrame(new_index_list).to_csv('../data/processed/index_list_combined_data.csv')

## Data dictionary

| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 0     | 'date'      | continuous (datetime) | Date standing for the month the value corresponds to |
| 1   | 'Betriebe_WZ08-10' | continuous (float) | Number of workshops producing animal feed or food|
| 2   | 'Betriebe_ep_WZ08-10' | categorical (nominal) | Finality of column above (e = final, p = provisional) |
| 3   | 'Beschäftigte_WZ08-10' | continuous (float) | Employees in see above|
| 4   | 'Beschäftigte_ep_WZ08-10' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 5   | 'Geleistete Arbeitsstunden_WZ08-10' | continuous (float) | Hours worked in see above|
| 6   | 'Geleistete Arbeitsstunden_ep_WZ08-10' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 7   | 'Bruttolohn- und -gehaltssumme_WZ08-10' | continuous (float) | Sum of wages in see above|
| 8   | 'Bruttolohn- und -gehaltssumme_ep_WZ08-10' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 9   | 'Umsatz_WZ08-10' | continuous (float) | ??|
| 10  | 'Umsatz_ep_WZ08-10' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 11  | 'Inlandsumsatz_WZ08-10' | continuous (float) | ??|
| 12  | 'Inlandsumsatz_ep_WZ08-10' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 13   | 'Auslandsumsatz_WZ08-10' | continuous (float) | ??|
| 14   | 'Auslandsumsatz_ep_WZ08-10' | categorical (nominal) | Finality of column above (e = final, p = provisional) |
| 15   | 'Auslandsumsatz mit der Eurozone_WZ08-10' | continuous (float) | ??|
| 16   | 'Auslandsumsatz mit der Eurozone_ep_WZ08-10' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 17   | 'Auslandsumsatz mit dem sonstigen Ausland_WZ08-10' | continuous (float) | ??|
| 18   | 'Auslandsumsatz mit dem sonstigen Ausland_ep_WZ08-10' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 19   | 'Betriebe_WZ08-108' | continuous (float) | Number of workshops producing other food|
| 20   | 'Betriebe_ep_WZ08-108' | categorical (nominal) | Finality of column above (e = final, p = provisional)|



| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 21   | 'Beschäftigte_WZ08-108' | continuous (float) | ??|
| 22   | 'Beschäftigte_ep_WZ08-108' | categorical (nominal) | Finality of column above (e = final, p = provisional) |
| 23   | 'Geleistete Arbeitsstunden_WZ08-108' | continuous (float) | ??|
| 24   | 'Geleistete Arbeitsstunden_ep_WZ08-108' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 25   | 'Bruttolohn- und -gehaltssumme_WZ08-108' | continuous (float) | ??|
| 26   | 'Bruttolohn- und -gehaltssumme_ep_WZ08-108' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 27   | 'Umsatz_WZ08-108' | continuous (float) | ??|
| 28   | 'Umsatz_ep_WZ08-108' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 29   | 'Inlandsumsatz_WZ08-108' | continuous (float) | ??|
| 30  | 'Inlandsumsatz_ep_WZ08-108' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 31  | 'Auslandsumsatz_WZ08-108' | continuous (float) | ??|
| 32  | 'Auslandsumsatz_ep_WZ08-108' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 33   | 'Auslandsumsatz mit der Eurozone_WZ08-108' | continuous (float) | ??|
| 34   | 'Auslandsumsatz mit der Eurozone_ep_WZ08-108' | categorical (nominal) | Finality of column above (e = final, p = provisional) |
| 35   | 'Auslandsumsatz mit dem sonstigen Ausland_WZ08-108' | continuous (float) | ??|
| 36   | 'Auslandsumsatz mit dem sonstigen Ausland_ep_WZ08-108' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 37   | 'Betriebe_WZ08-1082' | continuous (float) | Number of workshops producing sweets (no 'Dauerbackware')|
| 38   | 'Betriebe_ep_WZ08-1082' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 39   | 'Beschäftigte_WZ08-1082' | continuous (float) | ??|
| 40   | 'Beschäftigte_ep_WZ08-1082' | categorical (nominal) | Finality of column above (e = final, p = provisional)|

| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 41   | 'Geleistete Arbeitsstunden_WZ08-1082' | continuous (float) | ??|
| 42   | 'Geleistete Arbeitsstunden_ep_WZ08-1082' | categorical (nominal) | Finality of column above (e = final, p = provisional) |
| 43   | 'Bruttolohn- und -gehaltssumme_WZ08-1082' | continuous (float) | ??|
| 44   | 'Bruttolohn- und -gehaltssumme_ep_WZ08-1082' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 45   | 'Umsatz_WZ08-1082' | continuous (float) | ??|
| 46   | 'Umsatz_ep_WZ08-1082' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 47   | 'Inlandsumsatz_WZ08-1082' | continuous (float) | ??|
| 48   | 'Inlandsumsatz_ep_WZ08-1082' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 49   | 'Auslandsumsatz_WZ08-1082' | continuous (float) | ??|
| 50  | 'Auslandsumsatz_ep_WZ08-1082' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 51   | 'Auslandsumsatz mit der Eurozone_WZ08-1082' | continuous (float) | ??|
| 52   | 'Auslandsumsatz mit der Eurozone_ep_WZ08-1082' | categorical (nominal) | Finality of column above (e = final, p = provisional) |
| 53   | 'Auslandsumsatz mit dem sonstigen Ausland_WZ08-1082' | continuous (float) | ??|
| 54   | 'Auslandsumsatz mit dem sonstigen Ausland_ep_WZ08-1082' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 55   | 'Betriebe_WZ08-1083' | continuous (float) | Number of workshops processing coffee and tea|
| 56   | 'Betriebe_ep_WZ08-1083' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 57   | 'Beschäftigte_WZ08-1083' | continuous (float) | ??|
| 58   | 'Beschäftigte_ep_WZ08-1083' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 59   | 'Geleistete Arbeitsstunden_WZ08-1083' | continuous (float) | ??|
| 60  | 'Geleistete Arbeitsstunden_ep_WZ08-1083' | categorical (nominal) | Finality of column above (e = final, p = provisional)|

| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 61   | 'Bruttolohn- und -gehaltssumme_WZ08-1083' | continuous (float) | ??|
| 62   | 'Bruttolohn- und -gehaltssumme_ep_WZ08-1083' | categorical (nominal) | Finality of column above (e = final, p = provisional) |
| 63   | 'Umsatz_WZ08-1083' | continuous (float) | ??|
| 64   | 'Umsatz_ep_WZ08-1083' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 65   | 'Inlandsumsatz_WZ08-1083' | continuous (float) | ??|
| 66   | 'Inlandsumsatz_ep_WZ08-1083' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 67   | 'Auslandsumsatz_WZ08-1083' | continuous (float) | ??|
| 68   | 'Auslandsumsatz_ep_WZ08-1083' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 69   | 'Auslandsumsatz mit der Eurozone_WZ08-1083' | continuous (float) | ??|
| 70  | 'Auslandsumsatz mit der Eurozone_ep_WZ08-1083' | categorical (nominal) | Finality of column above (e = final, p = provisional)|
| 71   | 'Auslandsumsatz mit dem sonstigen Ausland_WZ08-1083' | continuous (float) | ??|
| 72   | 'Auslandsumsatz mit dem sonstigen Ausland_ep_WZ08-1083' | categorical (nominal) | Finality of column above (e = final, p = provisional) |
| 73   | 'ErzPr_Nahrungsmittel und Futtermittel' | continuous (float) | ??|
| 74   | 'ErzPr_ Sonstige Nahrungsmittel (ohne Getränke)' | continuous (float) | |
| 75   | 'ErzPr_ Süßwaren (ohne Dauerbackwaren)' | continuous (float) | ??|
| 76   | 'ErzPr_ Schokoladen und andere Süßwaren' | continuous (float) | |
| 77   | 'ErzPr_ Schokolade u.a. kakaoh. Lebensm.zub.,in Verp.>2kg' | continuous (float) | ??|
| 78   | 'ErzPr_ Schokolade u.a. kakaoh. Leb.m.zuber.,in Verp.<=2kg' | continuous (float) | |
| 79   | 'ErzPr_ Süßwaren oh. Kakaogeh. (einschl.weißer Schokolade)' | continuous (float) | ??|
| 80  | 'ErzPr_ Kaffee und Tee, Kaffee-Ersatz' | continuous (float) | |

| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 81   | 'Kaffee und Tee, Kaffee-Ersatz' | continuous (float) | ??|
| 82   | 'ErzPr_ Kaffee, entkoffeiniert oder geröstet' | continuous (float) |  |
| 83   | 'ErzPr_ Kaffeemittel m. -geh.,Auszüge,Essenz.,Konzentrate' | continuous (float) | ??|
| 84   | 'Grüner Tee (n. ferment.), schwarzer Tee (ferment.)' | continuous (float) ||
| 85   | 'ErzPr_ Kräutertees' | continuous (float) | ??|
| 86   | 'EinfPr_Nahrungsmittel und Futtermittel' | continuous (float) | |
| 87   | 'EinfPr_ Sonstige Nahrungsmittel (ohne Getränke)' | continuous (float) | ??|
| 88   | 'EinfPr_ Süßwaren (ohne Dauerbackwaren)' | continuous (float) | |
| 89   | 'EinfPr_ Kaffee und Tee, Kaffee-Ersatz' | continuous (float) | ??|
| 90  | 'VPI_Nahrungsmittel und alkoholfreie Getränke' | continuous (float) | |
| 91   | 'VPI_ Nahrungsmittel' | continuous (float) | ??|
| 92   | 'VPI_ Zucker, Marmelade, Honig und andere Süßwaren' | continuous (float) |  |
| 93   | 'VPI_ Schokoladen' | continuous (float) | ??|
| 94   | 'VPI_ Süßwaren' | continuous (float) ||
| 95   | 'VPI_ Alkoholfreie Getränke' | continuous (float) | ??|
| 96   | 'VPI_ Kaffee, Tee und Kakao' | continuous (float) | |
| 97   | 'VPI_ Kaffee und Ähnliches' | continuous (float) | ??|
| 98   | 'VPI_ Tee und teeähnliche Erzeugnisse' | continuous (float) | |
| 99   | 'VPI_ Kakaopulver oder Ähnliches' | continuous (float) | ??|
| 100  | 'Wert der zum Absatz bestimmten Produktion_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float) | |

| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 101    | 'Wert der zum Absatz bestimmten Produktion_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 102    | 'Gewicht der zum Absatz bestimmten Produktion_t_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 103    | 'Gewicht der zum Absatz bestimmten Produktion_t_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' |categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 104    | 'Gewicht der zum Absatz bestimmten Produktion_kg_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 105    | 'Gewicht der zum Absatz bestimmten Produktion_kg_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' |categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 106   | 'Menge der zum Absatz bestimmten Produktion_stueck_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 107    | 'Menge der zum Absatz bestimmten Produktion_stueck_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' |categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 108    | 'Menge der zum Absatz bestimmten Produktion_jewME_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 109    | 'Menge der zum Absatz bestimmten Produktion_jewME_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 110   | 'Unternehmen mit zum Absatz bestimmter Produktion_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 111    | 'Unternehmen mit zum Absatz bestimmter Produktion_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' |categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 112    | 'Gewicht der Gesamtproduktion_t_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 113    | 'Gewicht der Gesamtproduktion_t_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 114   | 'Gewicht der Gesamtproduktion_kg_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 115  | 'Gewicht der Gesamtproduktion_kg_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 116   | 'Menge der Gesamtproduktion_stueck_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 117   | 'Menge der Gesamtproduktion_stueck_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 118    | 'Menge der Gesamtproduktion_jewME_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||
| 119    | 'Menge der Gesamtproduktion_jewME_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 120   | 'Unternehmen mit Produktion_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | continuous (float)||

| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 121    | 'Unternehmen mit Produktion_ep_Leb-, Honigkuchen, Kekse, Kleingebäck, Waffeln' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 122    | 'Wert der zum Absatz bestimmten Produktion_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' |continuous (float) ||
| 123    | 'Wert der zum Absatz bestimmten Produktion_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 124    | 'Gewicht der zum Absatz bestimmten Produktion_t_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 125    | 'Gewicht der zum Absatz bestimmten Produktion_t_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 126   | 'Gewicht der zum Absatz bestimmten Produktion_kg_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 127    | 'Gewicht der zum Absatz bestimmten Produktion_kg_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 128    | 'Menge der zum Absatz bestimmten Produktion_stueck_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 129    | 'Menge der zum Absatz bestimmten Produktion_stueck_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' |categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 130   | 'Menge der zum Absatz bestimmten Produktion_jewME_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 131    | 'Menge der zum Absatz bestimmten Produktion_jewME_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 132    | 'Unternehmen mit zum Absatz bestimmter Produktion_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 133    | 'Unternehmen mit zum Absatz bestimmter Produktion_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 134   | 'Gewicht der Gesamtproduktion_t_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 135  | 'Gewicht der Gesamtproduktion_t_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 136   | 'Gewicht der Gesamtproduktion_kg_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 137   | 'Gewicht der Gesamtproduktion_kg_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 138    | 'Menge der Gesamtproduktion_stueck_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 139    | 'Menge der Gesamtproduktion_stueck_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 140   | 'Menge der Gesamtproduktion_jewME_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||

| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 141    | 'Menge der Gesamtproduktion_jewME_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 142    | 'Unternehmen mit Produktion_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | continuous (float)||
| 143    | 'Unternehmen mit Produktion_ep_Schokolade u.a. kakaohaltige Lebensmittelzubereit.' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 144    | 'Wert der zum Absatz bestimmten Produktion_Kaffee, geröstet' | continuous (float)||
| 145    | 'Wert der zum Absatz bestimmten Produktion_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 146   | 'Gewicht der zum Absatz bestimmten Produktion_t_Kaffee, geröstet' | continuous (float)||
| 147    | 'Gewicht der zum Absatz bestimmten Produktion_t_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 148    | 'Gewicht der zum Absatz bestimmten Produktion_kg_Kaffee, geröstet' | continuous (float)||
| 149    | 'Gewicht der zum Absatz bestimmten Produktion_kg_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 150   | 'Menge der zum Absatz bestimmten Produktion_stueck_Kaffee, geröstet' | continuous (float)||
| 151    | 'Menge der zum Absatz bestimmten Produktion_stueck_ep_Kaffee, geröstet' |categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 152    | 'Menge der zum Absatz bestimmten Produktion_jewME_Kaffee, geröstet' | continuous (float)||
| 153    | 'Menge der zum Absatz bestimmten Produktion_jewME_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 154   | 'Unternehmen mit zum Absatz bestimmter Produktion_Kaffee, geröstet' | continuous (float)||
| 155  | 'Unternehmen mit zum Absatz bestimmter Produktion_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 156   | 'Gewicht der Gesamtproduktion_t_Kaffee, geröstet' | continuous (float)||
| 157   | 'Gewicht der Gesamtproduktion_t_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 158    | 'Gewicht der Gesamtproduktion_kg_Kaffee, geröstet' | continuous (float)||
| 159    | 'Gewicht der Gesamtproduktion_kg_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 160   | 'Menge der Gesamtproduktion_stueck_Kaffee, geröstet' |continuous (float) ||

| Column number | Column name | Data type | Description |
| ----------- | ----------- | --------------- | ----------------- |
| 161    | 'Menge der Gesamtproduktion_stueck_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 162    | 'Menge der Gesamtproduktion_jewME_Kaffee, geröstet' | continuous (float)||
| 163    | 'Menge der Gesamtproduktion_jewME_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 164    | 'Unternehmen mit Produktion_Kaffee, geröstet' | continuous (float)||
| 165    | 'Unternehmen mit Produktion_ep_Kaffee, geröstet' | categorical (nominal) |Finality of column above (e = final, p = provisional)|
| 166   | 'PCOCOUSDM' | continuous (float)|Global cacao price|
| 167    | 'PCOFFROBUSDM' | continuous (float)|Global coffee robusta price|
| 168    | 'Ferrero' | continuous (float)| Share price|
| 169    | 'Mondelez' | continuous (float)| Share price|
| 170   | 'LindtSpruengli' | continuous (float)| Share price|